File: samples/setup/Setup.WPF/Bootstrapper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows;
using Caliburn.Micro;
using Setup.WPF.ViewModels;

namespace Setup.WPF
{
    public class Bootstrapper : BootstrapperBase
    {
        private SimpleContainer container;

        public Bootstrapper()
        {
            Initialize();
        }

        protected override void Configure()
        {
            container = new SimpleContainer();

            container.Singleton<IWindowManager, WindowManager>();

            // container.PerRequest<ShellViewModel>();
            GetType().Assembly.GetTypes().Where(type => type.IsClass)
     .Where(type => type.Name.EndsWith("ViewModel"))
     .ToList()
     .ForEach(viewModelType => container.RegisterPerRequest
         (viewModelType, viewModelType.ToString(), viewModelType));
        }

        protected override async void OnStartup(object sender, StartupEventArgs e)
        {
           await DisplayRootViewFor<ShellViewModel>();
        }

        protected override object GetInstance(Type service, string key)
        {
            return container.GetInstance(service, key);
        }

        protected override IEnumerable<object> GetAllInstances(Type service)
        {
            return container.GetAllInstances(service);
        }

        protected override void BuildUp(object instance)
        {
            container.BuildUp(instance);
        }
    }
}

File: samples/setup/Setup.WPF.Core.VB/Bootstrapper.vb
Imports Caliburn.Micro

Public Class Bootstrapper
    Inherits BootstrapperBase

    Private container As SimpleContainer

    Public Sub New()
        Initialize()
    End Sub

    Protected Overrides Sub Configure()
        container = New SimpleContainer()

        container.Singleton(Of IWindowManager, WindowManager)()

        container.PerRequest(Of ShellViewModel)()
    End Sub

    Protected Overrides Sub OnStartup(sender As Object, e As StartupEventArgs)
        DisplayRootViewFor(Of ShellViewModel)()
    End Sub

    Protected Overrides Function GetInstance(service As Type, key As String) As Object
        Return container.GetInstance(service, key)
    End Function

    Protected Overrides Function GetAllInstances(service As Type) As IEnumerable(Of Object)
        Return container.GetAllInstances(service)
    End Function

    Protected Overrides Sub BuildUp(instance As Object)
        container.BuildUp(instance)
    End Sub
End Class
